eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/RSMergeWriter.java

```diff
@@ -192,6 +193,18 @@ public void writeRSMerge(SubmittedVariantOperationEntity currentOperation)
                 getMergeDestinationAndMergees(mergeCandidates);
         ClusteredVariantEntity mergeDestination = mergeDestinationAndMergees.getLeft();
 
+        // check if any variant with same hash as mergeDestination already exist in DB
+        ClusteredVariantEntity existingClusteredVariantEntity = getClusteredVariantEntityWithHash(mergeDestination);
+        if (existingClusteredVariantEntity != null && existingClusteredVariantEntity.getAccession()!=mergeDestination.getAccession()) {
+            if (mergeDestination.getAccession() == ClusteredVariantMergingPolicy.prioritise(mergeDestination.getAccession(),
+                    existingClusteredVariantEntity.getAccession()).accessionToKeep) {
+                merge(mergeDestination, existingClusteredVariantEntity, currentOperation);
+            } else {
+                merge(existingClusteredVariantEntity, mergeDestination, currentOperation);
+                mergeDestination = existingClusteredVariantEntity;
```

Comment:
I don't think this part is necessary. If the existing cluster variant entity document is the merge destination then the norm code bellow would work.